### PR TITLE
[OSCD] Tests added for T1497.001

### PR DESCRIPTION
**Pull request by haresudhan**
**Details:**
Added tests for T1497.001 - Virtualization/Sandbox Evasion: System Checks

**Testing:**
Done locally on Windows, CentOS, Ubuntu and MacOS machines. 

**Associated Issues:**
None

**References**
1. https://attack.mitre.org/techniques/T1497/001/
2. https://evasions.checkpoint.com/techniques/macos.html
3. https://unit42.paloaltonetworks.com/unit42-oilrig-targets-middle-eastern-government-adds-evasion-techniques-oopsie/

### CODE REVIEW

**Comment by clr2of8** [comment]
@rc-grey I reviewed the Windows version of this test and it looks great. T1497.001 is a good fit as well. Could you review the Linux/macOS versions?

File: atomics/T1497.001/T1497.001.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  description: |
+    systemd-detect-virt detects execution in a virtualized environment.
+    At boot, dmesg stores a log if a hypervisor is detected.
+  supported_platforms:
+    - linux
+  executor:
+    name: sh
+    elevation_required: false 
+    command: | 
+      if (systemd-detect-virt || dmesg | grep -i "hypervisor detected") then echo "Virtualization Environment detected"; fi;
```
Comment (rc-grey) [inline atomics/T1497.001/T1497.001.yaml:15]: Id prefer the fallback here being similar to the `macos` method, using `dmidecode` to detect manufacturer. The kernel ring buffer can roll, be turned off, or cleared.

Example to detect VMWare
`sudo dmidecode | egrep -i 'manufacturer|product|vendor' | grep -i VMWare`

**Review by rc-grey — CHANGES_REQUESTED** [review]

File: atomics/T1497.001/T1497.001.yaml (hunk 1 above)
Comment (haresudhan) [reply atomics/T1497.001/T1497.001.yaml]: Sure @rc-grey . Thanks. Changed it.

**Review by rc-grey — APPROVED** [review]

File: atomics/T1497.001/T1497.001.yaml (hunk 1 above)
Comment (rc-grey) [reply atomics/T1497.001/T1497.001.yaml]: Thanks for the contribution!

**Pull request merged**